x-pack/legacy/plugins/epm/server/lib/kibana/index_pattern/install.ts

```diff
@@ -5,16 +5,90 @@
  */
 
 import { SavedObjectsClientContract } from 'kibana/server';
+import { Field } from '../../fields/field';
+import { loadFieldsFromYaml } from '../../elasticsearch/template/install';
+import * as Registry from '../../../registry';
 
-export async function installIndexPattern(
-  name: string,
+interface IndexPatternField extends Field {
+  searchable: boolean;
+  aggregatable: boolean;
+  readFromDocValues: boolean;
+}
+export async function installIndexPatterns(
+  pkgkey: string,
   savedObjectsClient: SavedObjectsClientContract
 ) {
-  await savedObjectsClient.create('index-pattern', getData(name));
+  const registryPackageInfo = await Registry.fetchInfo(pkgkey);
+  const fields = await loadFieldsFromYaml(registryPackageInfo);
+  const kibanaIndexPatternFields = makeKibanaIndexPatternFields(fields);
+  // TODO: separate out logs and metrics patterns
+  await savedObjectsClient.create(
+    'index-pattern',
```

Comment:
Mind updating this to use `KibanaAssetType.indexPattern`? I realize it's not added by this PR, but noticed it while reading. We can update later if you'd rather.